evennia/contrib/mail.py

```diff
@@ -0,0 +1,194 @@
+                        old_message.tags.remove("u", category="mail")
+                        old_message.tags.add("r", category="mail")
+                        return
+                    else:
+                        self.caller.msg("Message does not exist.")
+                        return
+
+            else:
+                if self.rhs:
+                    if "/" in self.rhs:
+                        subject, body = self.rhs.split("/", 1)
+                    else:
+                        body = self.rhs
+                    self.send_mail(self.lhslist, subject, body, self.caller)
+                else:
+                    message = self.get_all_mail()[int(self.lhs) - 1]
+
+                    messageForm = []
+                    if message:
+                        messageForm.append(HEAD_CHAR * 78)
+                        messageForm.append("{wFrom:{n %s" % (message.senders[0].key))
+                        messageForm.append("{wSent:{n %s" % message.db_date_created.strftime("%m/%d/%Y %H:%M:%S"))
+                        messageForm.append("{wSubject:{n %s" % message.header)
+                        messageForm.append(SUB_HEAD_CHAR * 78)
+                        messageForm.append(message.message)
+                        messageForm.append(HEAD_CHAR * 78)
+                    self.caller.msg("\n".join(messageForm))
+                    message.tags.remove("u", category="mail")
+                    message.tags.add("o", category="mail")
+
+        else:
+            messages = self.get_all_mail()
+
+            if messages:
+                table = evtable.EvTable("{wID:", "{wFrom:", "{wSubject:", "{wDate:{n", "{wSta:{n",
+                                        table=None, border="header", header_line_char=SUB_HEAD_CHAR, width=78)
+                index = 1
+                for message in messages:
+                    table.add_row(index, message.senders[0], message.header,
+                                  message.db_date_created.strftime("%m/%d/%Y"),
+                                  str(message.db_tags.last().db_key.upper()))
+                    index += 1
+
+                table.reformat_column(0, width=6)
+                table.reformat_column(1, width=17)
+                table.reformat_column(2, width=34)
+                table.reformat_column(3, width=13)
+                table.reformat_column(4, width=7)
+
+                self.caller.msg(HEAD_CHAR * 78)
+                self.caller.msg(table)
+                self.caller.msg(HEAD_CHAR * 78)
+            else:
+                self.caller.msg("Sorry, you don't have any messages.  What a pathetic loser!")
+
+    def get_all_mail(self):
+        mail_messages = Msg.objects.get_by_tag(category="mail")
+        messages = []
+        messages = [m for m in mail_messages if self.caller.player in m.receivers]
+        return messages
+
+    def send_mail(self, recipients, subject, message, caller):
```

Comment:
This needs a docstring in Google format. See the [Evennia code style guide](https://github.com/evennia/evennia/blob/master/CODING_STYLE.md#function--method-docstrings).